modules/sdk-coin-avaxc/src/avaxc.ts

```diff
@@ -360,15 +225,429 @@ export class AvaxC extends BaseCoin {
   }
 
   /**
-   * Builds a funds recovery transaction without BitGo.
-   * We need to do three queries during this:
+   * Check whether gas limit passed in by user are within our max and min bounds
+   * If they are not set, set them to the defaults
+   * @param userGasLimit user defined gas limit
+   * @returns the gas limit to use for this transaction
+   */
+  setGasLimit(userGasLimit?: number): number {
+    if (!userGasLimit) {
+      return ethGasConfigs.defaultGasLimit;
+    }
+    const gasLimitMax = ethGasConfigs.maximumGasLimit;
+    const gasLimitMin = ethGasConfigs.minimumGasLimit;
+    if (userGasLimit < gasLimitMin || userGasLimit > gasLimitMax) {
+      throw new Error(`Gas limit must be between ${gasLimitMin} and ${gasLimitMax}`);
+    }
+    return userGasLimit;
+  }
+
+  /**
+   * Check whether the gas price passed in by user are within our max and min bounds
+   * If they are not set, set them to the defaults
+   * @param userGasPrice user defined gas price
+   * @returns the gas price to use for this transaction
+   */
+  setGasPrice(userGasPrice?: number): number {
+    if (!userGasPrice) {
+      return ethGasConfigs.defaultGasPrice;
+    }
+
+    const gasPriceMax = ethGasConfigs.maximumGasPrice;
+    const gasPriceMin = ethGasConfigs.minimumGasPrice;
+    if (userGasPrice < gasPriceMin || userGasPrice > gasPriceMax) {
+      throw new Error(`Gas price must be between ${gasPriceMin} and ${gasPriceMax}`);
+    }
+    return userGasPrice;
+  }
+
+  /**
+   * Make a query to Snowtrace for information such as balance, token balance, solidity calls
+   * @param query {Object} key-value pairs of parameters to append after /api
+   * @returns {Object} response from Snowtrace
+   */
+  async recoveryBlockchainExplorerQuery(query: Record<string, string>): Promise<any> {
+    const token = common.Environments[this.bitgo.getEnv()].snowtraceApiToken;
+    if (token) {
+      query.apikey = token;
+    }
+    const response = await request.get(common.Environments[this.bitgo.getEnv()].snowtraceBaseUrl + '/api').query(query);
+
+    if (!response.ok) {
+      throw new Error('could not reach Snowtrace');
+    }
+
+    if (response.body.status === '0' && response.body.message === 'NOTOK') {
+      throw new Error('Snowtrace rate limit reached');
+    }
+    return response.body;
+  }
+
+  /**
+   * Queries public block explorer to get the next nonce that should be used for
+   * the given AVAXC address
+   * @param address
+   * @returns {*}
+   */
+  async getAddressNonce(address: string): Promise<number> {
+    // Get nonce for backup key (should be 0)
+    let nonce = 0;
+
+    const result = await this.recoveryBlockchainExplorerQuery({
```

Comment:
I'm a bit curious why we can't just go to the fullnode for this sort of information via the IMS. IMS supports the eth_getTransactionCount RPC call, though admittedly I'm less informed on AVAX

Reply:
I followed the ETH approach. I think since it's for BitGo-less support, we want to avoid relying on our infrastructure.

Reply:
Ah that makes sense then, right, this is SDK I'm reviewing not WP 😮‍💨